corehq/apps/data_dictionary/views.py

```diff
@@ -424,42 +444,51 @@ def _process_bulk_upload(bulk_file, domain):
     errors = []
     import_fhir_data = toggles.FHIR_INTEGRATION.enabled(domain)
     fhir_resource_type_by_case_type = {}
-    expected_columns_in_prop_sheet = 5
     data_type_map = {t.label: t.value for t in CaseProperty.DataType}
 
-    if import_fhir_data:
-        expected_columns_in_prop_sheet = 7
-
     worksheets = []
     allowed_value_info = {}
     prop_row_info = {}
     seen_props = defaultdict(set)
     missing_valid_values = set()
     with open_any_workbook(filename) as workbook:
         for worksheet in workbook.worksheets:
-            if worksheet.title.endswith(ALLOWED_VALUES_SHEET_SUFFIX):
-                case_type = worksheet.title[:-len(ALLOWED_VALUES_SHEET_SUFFIX)]
-                allowed_value_info[case_type] = defaultdict(dict)
-                prop_row_info[case_type] = defaultdict(list)
-                for (i, row) in enumerate(itertools.islice(worksheet.iter_rows(), 1, None), start=2):
-                    row_len = len(row)
-                    if row_len < 1:
-                        # simply ignore any fully blank rows
-                        continue
-                    if row_len < 3:
-                        # if missing value or description, fill in "blank"
-                        row += [Cell(value='') for __ in range(3 - row_len)]
-                    row = [cell.value if cell.value is not None else '' for cell in row]
-                    prop_name, allowed_value, description = [str(val) for val in row[0:3]]
-                    if allowed_value and not prop_name:
-                        msg_format = _('Error in valid values for case type {}, row {}: missing case property')
-                        msg_val = msg_format.format(case_type, i)
-                        errors.append(msg_val)
-                    else:
-                        allowed_value_info[case_type][prop_name][allowed_value] = description
-                        prop_row_info[case_type][prop_name].append(i)
-            else:
+            if not worksheet.title.endswith(ALLOWED_VALUES_SHEET_SUFFIX):
                 worksheets.append(worksheet)
+                continue
+
+            case_type = worksheet.title[:-len(ALLOWED_VALUES_SHEET_SUFFIX)]
+            allowed_value_info[case_type] = defaultdict(dict)
+            prop_row_info[case_type] = defaultdict(list)
+            column_headings = []
+            for (i, row) in enumerate(itertools.islice(worksheet.iter_rows(), 0, None), start=1):
+                if i == 1:
```

Comment:
This is a small personal thing, so no need to update if you don't want to: I would change this to
```python
is_header_row = i == 1
if is_header_row:
    ...
```
this way it makes it clear that we're going into this if branch when looking at the header row. One might argue that the variable assignment is useless, but it gives the `i == i` meaning and makes the code a bit easier to read.

Reply:
I agree that it's always good to try and minimize the use of magic numbers. I have replaced 1 with a `HEADER_ROW_INDEX` global constant ([65919d1](https://github.com/dimagi/commcare-hq/pull/33259/commits/65919d13fec2584c38b352f5e9c5ec96e4222ba2)).